src/swsssdk/configdb.py

```diff
@@ -198,14 +200,44 @@ def mod_entry(self, table, key, data):
         if data == None:
             client.delete(_hash)
         else:
-            client.hmset(_hash, self.__typed_to_raw(data))
+            # We allow deletion of field by setting them to None, __typed_to_raw()
+            # will fill keys to be deleted in dkeys arg.
+            dkeys = dict()
```

Comment:
Same as above,  I feel we should use dkeys (rename to dfields) as return value from __typed_to_raw function.